### PR TITLE
fixing pom issues that prevented publishing

### PR DESCRIPTION
**Pull request by lbergelson**
* fixing two issues that prevented us from publishing to maven central
  - fixing sourceJar typo -> sourcesJar
  - configure the pom for all artifact filters
* partial fix for https://github.com/broadinstitute/gatk/issues/5212

### CODE REVIEW

**Comment by lbergelson** [comment]
I'm not sure if there's a cleaner way to specify the pom without having to set it in so many places.  If anyone has a good suggestion let me know...

File: build.gradle [hunk 1]
```diff
@@ -623,9 +623,9 @@ task testUtilsJavadocJar(type: Jar, dependsOn: testUtilsJavadoc){
     from "$docBuildDir/testUtilsJavadoc"
 }
 
-task testUtilsSourceJar(type: Jar){
+task testUtilsSourcesJar(type: Jar){
     baseName = "$project.name-test-utils"
-    classifier = 'source'
+    classifier = 'sources'
```
Comment (cmnbroad) [inline build.gradle]: Wow - was this just silently not working ?

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariantsIntegrationTest.java [hunk 2]
```diff
@@ -40,6 +41,17 @@ public void testSampleSelection() throws IOException {
...
+
+    @Test
+    public void testBrokenThing() throws IOException {
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        args.addVCF(new File("denovo_2-5pct_1-filtered.vcf"));
+        args.addReference(new File("/Users/louisb/data/Homo_sapiens_assembly19.fasta"));
+        args.addOutput(new File("out.vcf"));
+
+        runCommandLine(args);
+    }
```
Comment (cmnbroad) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariantsIntegrationTest.java]: Assuming inclusion of this is an oversight ?

Comment (lbergelson) [reply src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariantsIntegrationTest.java]: Ack, yes, that is definitely an oversight...

File: build.gradle (hunk 1 above)
Comment (lbergelson) [reply build.gradle]: It was loudly not working when we tried to do a maven release.

**Pull request merged**